### PR TITLE
New PDF book added - Developing 2D Games with Unity

### PR DESCRIPTION
**Pull request by KiranJodhani**
## What does this PR do?
Added new book for Game Development using Unity3D Game Engine
### Description
This book was written for programmers interested in making video games with Unity. 
### Why is this valuable (or not)?
To learn game development
### How do we know it's really free?
Click to open book from any web browser - https://resources.unity.com/games/level-up-your-code-with-game-programming-patterns
### For book lists, is it a book? For course lists, is it a course? etc.
It's a PDF book
## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by eshellman** [comment]
Are you associated with this book? If it is free we would need the publisher or author's link, not yours.

**Comment by KiranJodhani** [comment]
I have updated new ebook and added publisher link. Thanks

File: books/free-programming-books-subjects.md [hunk 1]
```diff
@@ -290,9 +290,11 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [3D Math Primer for Graphics and Game Development](https://gamemath.com/book/intro.html) - Fletcher Dunn (HTML)
 * [Coding With Minecraft](https://turtleappstore.com/book/) - Al Sweigart
 * [Designing Virtual Worlds](https://mud.co.uk/richard/DesigningVirtualWorlds.pdf) - Richard A. Bartle (PDF)
+
```
Comment (eshellman) [inline books/free-programming-books-subjects.md:293]: remove this space

Comment (KiranJodhani) [reply books/free-programming-books-subjects.md]: Space removed sir
Thanks

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Comment by KiranJodhani** [comment]
Thank you!

**Pull request merged**